typed-racket-lib/typed-racket/base-env/base-special-env.rkt

```diff
@@ -313,7 +316,7 @@
    (-> Univ (-lst (-Syntax Univ)) (-Syntax Univ))]
   ;; ... or a Guide is ...
   [(make-template-identifier 'h-splice 'racket/private/template)
-   (-> Univ (Un (-val #f) (-Syntax Univ)) (Un (-val #f) (-Syntax Univ)) (-lst (-Syntax Univ)))]
+   (-> Univ (-pair -Symbol -Symbol) (Un (-val #f) (-Syntax Univ)) (-lst (-Syntax Univ)))]
```

Comment:
@rmculpepper  can you double-check this type?

In #690 you'd suggested the type `h-splice : Any (U Syntax #f) (U Syntax #f) -> (Listof Syntax)`

Maybe the 2nd argument should be a `-Stxish` [(link)](https://github.com/racket/typed-racket/blob/master/typed-racket-lib/typed-racket/types/abbrev.rkt#L100)?

Reply:
The second and third arguments can probably just be `Any`; they're just there for error reporting.

Reply:
@bennn I've pushed a new commit that widens these types.